packages/client/lib/sentinel/pub-sub-proxy.ts

```diff
@@ -0,0 +1,189 @@
+    }
+
+    options.socket.host = this.#node.host;
+    options.socket.port = this.#node.port;
+    
+
+    return RedisClient.create(options);
+  }
+
+  async changeNode(node: RedisNode) {
+    this.#node = node;
+    
+    if (this.hasListeners()) {
+        if (this.#pubSubNode) {
+            this.destroy();
+        }
+
+        await this.#initiatePubSubClient();
+    }
+  }
+
+  async close() {
+    if (this.#pubSubNode) {
+        const client = this.#pubSubNode.client;
+
+        /* probably less needed, as close() means we're probably also shutting down whoever holds this */
+        this.#channelsListeners = client.getPubSubListeners(PubSubType.CHANNELS);
+        this.#patternsListeners = client.getPubSubListeners(PubSubType.PATTERNS);
+
+        this.#pubSubNode = undefined;
+        return client.close();
+    }
+  }
+
+  destroy() {
+    if (this.#pubSubNode) {
+        const client = this.#pubSubNode.client;
+
+        this.#channelsListeners = client.getPubSubListeners(PubSubType.CHANNELS);
+        this.#patternsListeners = client.getPubSubListeners(PubSubType.PATTERNS);
+
+        this.#pubSubNode = undefined;
+        return client.destroy();
+    }
+  }
+
+  async #initiatePubSubClient() {
+    const client = this.#createClient()
+      
+    this.#pubSubNode = {
+      client: client,
+      connectPromise: client.connect()
+        .then(async client => {
+          if (this.#channelsListeners && this.#channelsListeners.size > 0) {
+            await client.extendPubSubListeners(PubSubType.CHANNELS, this.#channelsListeners)
+          }
+          if (this.#patternsListeners && this.#patternsListeners.size > 0) {
+            await client.extendPubSubListeners(PubSubType.PATTERNS, this.#patternsListeners);
+          }
+
+          this.#pubSubNode!.connectPromise = undefined;
+          return client;
+        })
+        .catch(err => {
+          this.#pubSubNode = undefined;
+          throw err;
+        })
+    };
+
+    return this.#pubSubNode.connectPromise!;
+  }
+
+  getPubSubClient()  {
+    if (!this.#pubSubNode) return this.#initiatePubSubClient();
+
+    return this.#pubSubNode.connectPromise ?? this.#pubSubNode.client;
+  }
+
+  async subscribe<T extends boolean = false>(
+    channels: string | Array<string>,
+    listener: PubSubListener<T>,
+    bufferMode?: T
+  ) {
+    const client = await this.getPubSubClient();
+    const resp = await client.SUBSCRIBE(channels, listener, bufferMode);
+
+    return resp;
+  }
+
+  async unsubscribe<T extends boolean = false>(
+    channels?: string | Array<string>,
+    listener?: PubSubListener<boolean>,
+    bufferMode?: T
+  ) {
+    const client = await this.getPubSubClient();
+    const resp = await client.UNSUBSCRIBE(channels, listener, bufferMode);
+
+    if (!client.isPubSubActive) {
+      client.destroy();
+      this.#pubSubNode = undefined;
+    }
+        
+    return resp;
+  }
+
+  async pSubscribe<T extends boolean = false>(
```

Comment:
is this pSubscribe/pUnsubscribe elsewhere because of camelCase? I would have expected this to follow the redis command name (psubscribe/punsubscribe)

Reply:
this is copying from rest of node api I believe.